### PR TITLE
Better responsive images

### PR DESCRIPTION
**Pull request by fabien-michel**
## Description

Rewrote of responsive images mode which now make use of `<picture>` HTML tag to support different pictures for small, medium and large screens and to allow to generate WebP alternative version.

User can now setup a specific optional alternative images for medium and large screen (small is the default one)
User can also specify optional viewport width (vw) for each alternative in order to help browser choose better version.
The plugin can now generate a WebP alternative version for each alternative in each size.

I've renamed two settings:
-   `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES` (enable responsive mode default value) become `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_ENABLED` 
-   `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_VIEWPORT_BREAKPOINTS`  (image sizes versions to generate)  become `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_SIZES`

There is new settings :
- `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_BREAKPOINT_MEDIUM` size of medium screen breakpoint. Default to `768`
- `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_BREAKPOINT_LARGE` size of large screen breakpoint. Default to `992`
- `DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_ALTERNATIVE_FORMAT_WEBP` generate WebP version. Default to `False`

For WebP alternative, [Pillow require](https://pillow.readthedocs.io/en/stable/installation.html#external-libraries) the libwebp library to be installed on the host system.

### Sample of HTML generated:
With an alternative picture (jpg) for large screen and default picture (png) for small and medium. The expected picture width is 100vw on small, default on medium and 40vw on large. With WebP alternative.

Settings:
```python
DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_ENABLED = True
DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_SIZES = [576, 768, 1024, 1280, 1366, 1536, 1920]
DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_BREAKPOINT_MEDIUM = 640
DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_BREAKPOINT_LARGE = 1024
DJANGOCMS_PICTURE_RESPONSIVE_IMAGES_ALTERNATIVE_FORMAT_WEBP = True
```

```html
<picture class="cms-plugin cms-plugin-16205">
  <source
    type="image/webp"
    media="(min-width: 1024px)"
    srcset="
      picture_A.jpg__576x576_crop_subsampling-2.webp    576w,
      picture_A.jpg__768x768_crop_subsampling-2.webp    768w,
      picture_A.jpg__1024x1024_crop_subsampling-2.webp 1024w,
      picture_A.jpg__1280x1280_crop_subsampling-2.webp 1280w,
      picture_A.jpg__1366x1366_crop_subsampling-2.webp 1366w,
      picture_A.jpg__1536x1536_crop_subsampling-2.webp 1536w,
      picture_A.jpg__1920x1920_crop_subsampling-2.webp 1920w
    "
    sizes="40vw"
  />

  <source
    type="image/webp"
    srcset="
      picture_B.png__576x576_crop_subsampling-2.webp    576w,
      picture_B.png__768x768_crop_subsampling-2.webp    768w,
      picture_B.png__1024x1024_crop_subsampling-2.webp 1024w,
      picture_B.png__1280x1280_crop_subsampling-2.webp 1280w,
      picture_B.png__1366x1366_crop_subsampling-2.webp 1366w,
      picture_B.png__1536x1536_crop_subsampling-2.webp 1536w,
      picture_B.png__1920x1920_crop_subsampling-2.webp 1920w
    "
    sizes="(min-width: 640px) and (max-width: 768px) 768px, 
           (min-width: 640px) and (max-width: 1024px) 1024px, 
           (min-width: 640px) and (max-width: 1280px) 1280px, 
           (min-width: 640px) and (max-width: 1366px) 1366px, 
           (min-width: 640px) and (max-width: 1536px) 1536px, 
           (min-width: 640px) 1920px, 
           100vw"
  />

  <source
    media="(min-width: 1024px)"
    srcset="
      picture_A.jpg__576x576_q85_crop_subsampling-2.jpg    576w,
      picture_A.jpg__768x768_q85_crop_subsampling-2.jpg    768w,
      picture_A.jpg__1024x1024_q85_crop_subsampling-2.jpg 1024w,
      picture_A.jpg__1280x1280_q85_crop_subsampling-2.jpg 1280w,
      picture_A.jpg__1366x1366_q85_crop_subsampling-2.jpg 1366w,
      picture_A.jpg__1536x1536_q85_crop_subsampling-2.jpg 1536w,
      picture_A.jpg__1920x1920_q85_crop_subsampling-2.jpg 1920w
    "
    sizes="40vw"
  />

  <img
    src="picture_B.png__1920x1281_crop_subsampling-2.png"
    alt=""
    srcset="
      picture_B.png__576x576_crop_subsampling-2.png    576w,
      picture_B.png__768x768_crop_subsampling-2.png    768w,
      picture_B.png__1024x1024_crop_subsampling-2.png 1024w,
      picture_B.png__1280x1280_crop_subsampling-2.png 1280w,
      picture_B.png__1366x1366_crop_subsampling-2.png 1366w,
      picture_B.png__1536x1536_crop_subsampling-2.png 1536w,
      picture_B.png__1920x1920_crop_subsampling-2.png 1920w
    "
    sizes="(min-width: 640px) and (max-width: 768px) 768px, 
           (min-width: 640px) and (max-width: 1024px) 1024px, 
           (min-width: 640px) and (max-width: 1280px) 1280px, 
           (min-width: 640px) and (max-width: 1366px) 1366px, 
           (min-width: 640px) and (max-width: 1536px) 1536px, 
           (min-width: 640px) 1920px, 
           100vw"
    class="align-center"
  />
</picture>
```

## Checklist

* [x] I have opened this pull request against ``master``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on 
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by fabien-michel** [comment]
What are the next steps about this PR ?
Have theses evolutions a place in djangocms-picture ?
If yes, are the proposed modifications acceptable or need changes ?
If no, should it be better to propose a fork (djangocms-picture-responsive) with this specific enhancement ?

May be it could be nice to talk about it with someone.

**Comment by marksweb** [comment]
@fabien-michel We've actually just been talking about this in one of our meetings. We have a meeting at 2pm UK time every other Friday. You're welcome to join, or post in the technical-committee or workgroup-pr-review channels on our slack; https://django-cms.org/slack

I don't use this app, so I've not commented on here. But hopefully with some visibility on slack, we might find some people who do use it & who can review this.

**Review by marksweb — CHANGES_REQUESTED** [review]
This review is taking me a while but here's some improvements to the docs

**Comment by i-salameh95** [comment]
I think this PR is so important, I was working on page insight for my website, and one of the suggestions is to use responsive images, I have use the settings by djangocms-picture provided OTB but another suggest said that: 
The size of the image doesn’t match the sizes attribute (max-width: 576px) 576px, (max-width: 768px) 768px, (max-width: 992px) 992px, 1000px. At a viewport of 320x427 the image was 294 pixels wide instead of the specified 576 (-49% difference). The affected viewports are 300x169-540x720, 580x326-740x987, 780x439-940x1253.

**_Try using sizes="(min-width: 1080px) 1000px, 95.53vw" instead._**
... that's why I search for this.

File: .github/workflows/publish-to-test-pypi.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Publish Python 🐍 distributions 📦 to TestPyPI
 on:
   push:
     branches:
-      - master
+      - better_responsive_images
```
Comment (marksweb) [inline .github/workflows/publish-to-test-pypi.yml:6]: Please revert this.

**Review by marksweb — CHANGES_REQUESTED** [review]

**Comment by fabien-michel** [comment]
Sorry, I was trying to make a fork and still have this PR alive.
I think the changes this PR adds make the UI more complex. It is hard for our editors here to understand how the responsiveness works and also how the image selection works according to screen size. 
So I think the work done in this PR needs to be "rethought" to a simpler UI before it would be nice to have it in djangocms-picture. 
I intend to make a new package "djangocms-picture-responsive" which contains this PR and I will try to maintain it. That's why I changed the branch name in ci config . 
Since it interferes too much with PR, I think it's better to close this PR for now.
Thanks for your time.

**Pull request closed without merging**